cmake/EthCompilerSettings.cmake

```diff
@@ -80,6 +80,14 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdata-sections -ffunction-sections -Wl,--gc-sections -fvisibility=hidden")
 			# Optimisation level
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
+			# Add runtime guards for memory allocation errors
+			# Enabled by default, but -O1, -O2, .. disables it.
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s ASSERTIONS=1")
```

Comment:
@chriseth @ekpyron since now we have release and non-release builds and tests on the CI, we could consider having these enabled for the non-release build so we would get testing of both.

wdyt?

Reply:
Is this relevant at all after the switch to wasm?

Reply:
As relevant as before, yes - the option will also affect wasm builds.


> @chriseth @ekpyron since now we have release and non-release builds and tests on the CI

We don't for the emscripten build, though, do we? We could of course add them for this purpose, but I'm still not convinced it's worth it...